ethergo/submitter/submitter.go

```
@@ -421,10 +419,58 @@
 		//nolint: wrapcheck
 		return parentTransactor.Signer(address, transaction)
 	}
+
+	// if dynamic gas estimation is not enabled, use cfg var gas_estimate as a gas limit default and do not run a pre-flight simulation
+	// since we do not need it to determine proper gas units
+	if !t.config.GetDynamicGasEstimate(int(chainID.Uint64())) {
+		transactor.GasLimit = t.config.GetGasEstimate(int(chainID.Uint64()))
+	} else {
+
+		// deepcopy the real transactor so we can use it for simulation
+		transactor_forGasEstimate := copyTransactOpts(transactor)
```

golangci-lint (1.62.2)

[warning] var-naming: don't use underscores in Go names; var transactor_forGasEstimate should be transactorForGasEstimate

(revive)